docs/guides/src/main/server/configuring-hostname.adoc

```diff
@@ -0,0 +1,177 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="Configuring the hostname"
+summary="Learn how to configure the frontend and backchannel endpoints exposed by Keycloak."
+includedOptions="hostname-* proxy">
+
+== Server Endpoints
+
+Keycloak exposes different endpoints to talk with applications as well as to allow accessing the administration console. These endpoints
+can be categorized into three main groups:
+
+* Frontend
+* Backend
+* Administration Console
+
+The base URL for each group has an important impact on how tokens are issued and validated, on how links are created for actions that require the user
+to be redirected to Keycloak (for example, when resetting password through email links), and, most importantly, how applications will
+discover these endpoints when fetching the OpenID Connect Discovery Document from `realms/{realm-name}/.well-known/openid-configuration`.
+
+=== Frontend
+
+The frontend endpoints are those accessible through a public domain and usually related to authentication/authorization flows that happen
+through the front-channel. For instance, when an SPA wants to authenticate their users it redirects them to the `authorization_endpoint` so that users
+can authenticate using their browsers through the front-channel.
+
+By default, when the hostname settings are not set, the base URL for these endpoints is based on the incoming request so that the HTTP scheme,
+host, port, and path, are the same from the request. The default behavior also has a direct impact on how the server is going to issue tokens given that the issuer is also based on
+the URL set to the frontend endpoints. If the hostname settings are not set, the token issuer will also be based on the incoming request and also lack consistency if the client is requesting tokens using different URLs.
+
+When deploying to production you usually want a consistent URL for the frontend endpoints and the token issuer regardless of how the request is constructed.
+In order to achieve this consistency, you can set either the `hostname` or the `hostname-url` options.
+
+Most of the time, it should be enough to set the `hostname` option in order to change only the *host* of the frontend URLs:
+
+<@kc.start parameters="--hostname=<host>"/>
+
+When using the `hostname` option the server is going to resolve the HTTP scheme, port, and path, automatically so that:
+
+* `https` scheme is used unless you set `http-strict-https=false`
```

Comment:
a) it's `hostname-strict-https=false`
b) we should finally unhide this configuration options as part of this PR (would then also solve #13353 and #13270 )